web-app/js/portal/filter/BooleanFilterPanel.js

```diff
@@ -41,8 +41,7 @@ Portal.filter.BooleanFilterPanel = Ext.extend(Portal.filter.BaseFilterPanel, {
     },
 
     _formatBoxLabel: function() {
-        var label = this.filter.label.split('_').join(' ').toTitleCase();
-        return label;
+        return this.filter.getDisplayLabel();
```

Comment:
I also envisage a similar class which takes a Filter and returns html for a human-readable representation for Step 3.